x-pack/plugins/security_solution/common/detection_engine/build_exceptions_filter.ts

```diff
@@ -0,0 +1,303 @@
+  EntryExists,
+} from '../../../lists/common';
+import { BooleanFilter, NestedFilter } from './types';
+import { hasLargeValueList } from './utils';
+
+type NonListEntry = EntryMatch | EntryMatchAny | EntryNested | EntryExists;
+interface ExceptionListItemNonLargeList extends ExceptionListItemSchema {
+  entries: NonListEntry[];
+}
+
+interface CreateExceptionListItemNonLargeList extends CreateExceptionListItemSchema {
+  entries: NonListEntry[];
+}
+
+export type ExceptionItemSansLargeValueLIsts =
+  | ExceptionListItemNonLargeList
+  | CreateExceptionListItemNonLargeList;
+
+export const chunkExceptions = (
+  exceptions: ExceptionItemSansLargeValueLIsts[],
+  chunkSize: number
+): ExceptionItemSansLargeValueLIsts[][] => {
+  if (exceptions.length === 0) {
+    return [];
+  } else if (exceptions.length <= chunkSize) {
+    return [exceptions];
+  } else {
+    const chunkedFilters: ExceptionItemSansLargeValueLIsts[][] = [];
+    for (let index = 0; index < exceptions.length; index += chunkSize) {
+      const exceptionsChunks = exceptions.slice(index, index + chunkSize);
+      chunkedFilters.push(exceptionsChunks);
+    }
+    return chunkedFilters;
+  }
+};
+
+export const buildExceptionItemFilter = (
+  exceptionItem: ExceptionItemSansLargeValueLIsts
+): BooleanFilter | NestedFilter => {
+  const { entries } = exceptionItem;
+
+  if (entries.length === 1) {
+    return createInnerAndClauses(entries[0]);
+  } else {
+    return {
+      bool: {
+        filter: entries.map((entry) => createInnerAndClauses(entry)),
+      },
+    };
+  }
+};
+
+export const createOrClauses = (
+  exceptionItems: ExceptionItemSansLargeValueLIsts[]
+): Array<BooleanFilter | NestedFilter> => {
+  return exceptionItems.map((exceptionItem) => buildExceptionItemFilter(exceptionItem));
+};
+
+export const buildExceptionFilter = ({
+  lists,
+  excludeExceptions,
+  chunkSize,
+}: {
+  lists: Array<ExceptionListItemSchema | CreateExceptionListItemSchema>;
+  excludeExceptions: boolean;
+  chunkSize: number;
+}): Filter | undefined => {
+  // Remove exception items with large value lists. These are evaluated
+  // elsewhere for the moment being.
+  const exceptionsWithoutLargeValueLists = lists.filter(
+    ({ entries }) => !hasLargeValueList(entries)
+  ) as ExceptionItemSansLargeValueLIsts[];
+
+  const exceptionFilter: Filter = {
+    meta: {
+      alias: null,
+      negate: excludeExceptions,
+      disabled: false,
+    },
+    query: {
+      bool: {
+        should: undefined,
+      },
+    },
+  };
+
+  if (exceptionsWithoutLargeValueLists.length === 0) {
+    return undefined;
+  } else if (exceptionsWithoutLargeValueLists.length <= chunkSize) {
+    const clause = createOrClauses(exceptionsWithoutLargeValueLists);
+    exceptionFilter.query.bool.should = clause;
+    return exceptionFilter;
+  } else {
+    const chunks = chunkExceptions(exceptionsWithoutLargeValueLists, chunkSize);
+
+    const filters: Filter[] = chunks.map((exceptionsChunk) => {
```

Comment:
I think you want to use templates with map like so:
```suggestion
    const filters = chunks.map<Filter>((exceptionsChunk) => {
```

Reply:
Brain fart there. Thanks for catching! Fixed it.